src/main/kotlin/com/hypto/iam/server/service/UsersService.kt

```diff
@@ -289,20 +289,37 @@ class UsersServiceImpl : KoinComponent, UsersService {
             throw BadRequestException("User - ${userHrn.resourceInstance} does not have login access")
         }
 
-        val identityGroup =
-            if (org.metadata != null) {
-                gson.fromJson(org.metadata.data(), IdentityGroup::class.java)
-            } else {
+        if (org.metadata != null) {
+            val identityGroup = gson.fromJson(org.metadata.data(), IdentityGroup::class.java)
+            identityProvider.setUserPassword(identityGroup, user.username, password)
+        } else {
+            txMan.wrap {
                 organizationRepo.update(
                     organizationId,
                     null,
                     null,
                     appConfig.cognito,
                 )
-                appConfig.cognito
+                createUserInIdentityProvider(
```

Comment:
Will this trigger email with passcode or is this after user receives passcode and clicks on the link and enters new password?

Reply:
After user receives passcode and clicks on it.